### DISCUSSION TITLE
Sample preprocessing in R

### DISCUSSION THREAD

**Discussion by MayaCyTOFnewbie** [ImmuneDynamics/Spectre#17] [Q&A] [answered]
Hi,

I would like to run all the analysis with R, without data preparation and export from FlowJo. I was wandering if there is a recommended tutorial to do so?

Thanks!

Maya

**Comment by tomashhurst** [comment, marked as answer]
Hi Maya, that's certainly doable! We don't have _our own_ tutorials on the the data cleanup etc in R (which is just a reflection of our personal preference to do the cleanup etc in FlowJo), but there are a number of other tutorials out there. On our [CyTOF resources page](https://immunedynamics.io/resources/mass) there is a link to a chapter that covers tools which can facilitate this in R: **Data-Driven Flow Cytometry Analysis (Sherrie Wang, Ryan R. Brinkman)**. That should give a nice run down of where to find resources for each relevant step. In particular, **flowAI** and **flowClean** will likely be the most help (they should work on both flow and mass cytometry data). If you get stuck then let us know.

Tom

**Reply by MayaCyTOFnewbie** [reply]
Hi Tom,

thank you for the quick reply! I decided to run the cleanup stage in FlowJo as well 

Best,
Maya
